cultivation/scripts/literature/docinsight_client.py

```
@@ -0,0 +1,101 @@
+            if 'timeout' in inspect.signature(requests.post).parameters:
+                post_kwargs['timeout'] = timeout
+            resp = requests.post(url, **post_kwargs)
+            if resp.status_code != 200:
+                raise DocInsightAPIError(f"start_research failed: {resp.status_code} {resp.text}")
+            data = resp.json()
+            job_id = data.get("job_id")
+            if not job_id:
+                raise DocInsightAPIError(f"No job_id returned: {data}")
+            return job_id
+        except requests.Timeout as e:
+            raise DocInsightTimeoutError(f"start_research timed out: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)